src/sentry_core.c

```diff
@@ -377,10 +378,11 @@ sentry__capture_event(sentry_value_t event)
     sentry_envelope_t *envelope = NULL;
 
     bool was_captured = false;
+    bool was_sent = false;
```

Comment:
while updating events i noticed this had a subtle bug: capture_event returned the event ID even if the event failed to send due to sampling and/or other internal issues. this fix takes care of sampling, however `sentry__capture_envelope` can silently fail due to missing user consent. i'm not sure if this was a deliberate decision, ie we do not want to let SDK users know when user consent has not been provided or if this was just overlooked.

i've made note of this as a TODO in the tests, but any insight on this would be great.

Reply:
my guess would be overlooked or too difficult for a first version (and second version never arrived)